src/connector/src/source/filesystem/opendal_source/mod.rs

```diff
@@ -53,6 +53,9 @@ pub struct GcsProperties {
 
     #[serde(flatten)]
     pub unknown_fields: HashMap<String, String>,
+
+    #[serde(rename = "decompression_format")]
+    pub decompression_format: Option<String>,
```

Comment:
Shall we make `decompression_format` a enum? See https://github.com/risingwavelabs/risingwave/blob/267e6c9410c13a23fe064c00ae46d9163e7de30e/src/common/src/config.rs#L558 as an example.

Reply:
SGTM

Reply:
Oh a bit hard, `WithOptions` can only be derived for structs.